### ISSUE TITLE
Make Jamulus iOS compatible with Qt6

### ISSUE THREAD

**Issue by ann0see** [jamulussoftware/jamulus#3406]
**Describe the bug**

Even though Jamulus can be compiled by Qt6 and it runs, there are multiple UI issues. 
Also the app hangs and crashes in multiple places. 

1. The chat window doesn't have the correct proportions. Especially if turning the device around the chat window overflows the screen (screenshot will follow)
2. Tapping the chat window Close button hangs and then crashes the app on the next tap. 
3. Changing the language in settings doesn't apply and the app hangs. 
4. Further bugs to be investigated

Related: https://github.com/jamulussoftware/jamulus/issues/2711

**Operating system**

iOS 17

**Version of Jamulus**

Latest version from main

**Additional context**

Try compiling Jamulus with Qt6 for iOS.

**Comment by ann0see** [comment]
Pasting https://forum.qt.io/topic/15326/solved-qlayout-items-not-expanding-on-resize/4 here. Maybe related to the crash of the languages change.

**Comment by ann0see** [comment]
@softins do you own an iPad?
I'd like to get the app on TestFlight to find further bugs.

**Comment by softins** [comment]
Yes, and have Testflight on it. iOS 16.7, which I think is the latest it will take (2016 model).

**Comment by ann0see** [comment]
Thanks. I'll try to work out how to get the app on TestFlight in the following days. It would be great if you could confirm if the mentioned issues all also occur on iPadOS

**Comment by ann0see** [comment]
Hotfix for 2 is probably not using the close() signal but hide() instead.

**Comment by ann0see** [comment]
Hotfix for 3 is conditionally getting rid of the warning message on iOS.

**Comment by ann0see** [comment]
Ok. Also the Qt5 build hangs with the language change. So that's not a new bug.
Qt5 doesn't even open the about dialog.
However, there's no need to turn the screen to get a working, non overflowing screen on Qt5.

**Comment by ann0see** [comment]
Changing the skin to compact, Bar (narrow) makes the app somewhat usable.

**Comment by ann0see** [comment]
Ok. The mixer board issue comes from the "Setup your audio,... message". I'll change it to be brief on iOS.

**Comment by ann0see** [comment]
We could force the GD_SLIMFADER design on iOS...
Hiding the text is not so easy. The issue still remains, if .hide() is applied on the text. However, if it is removed from the Qt created code, it's ok. 
Potentially also the stylesheet for the fancy design could be adapted.

**Comment by ann0see** [comment]
Language bug might be fixable by using a Qt::QueuedConnection.

https://bugreports.qt.io/browse/QTBUG-64577

**Comment by ann0see** [comment]
Further error:
5. The connect window has wrong proportions but can be fixed by turning the screen around

Probably that's related to the same issue as the mixer board: The table might be too wide

**Comment by ann0see** [comment]
6. Mute myself bar is too wide

**Comment by ann0see** [comment]
AppStore connect requests a usage description for camera access. This could be related to file access of the app.

**Comment by ann0see** [comment]
7. The save mixer channel dialog overflows the screen and doesn't use an iOS integrated one - this dialog type doesn't exist on iOS. We'd need to specify a name before saving, I suppose. Then maybe invoke the share sheet?

**Comment by ann0see** [comment]
Concerning the UI issues, I'd like to start with the mixer board.

I think all of the issues are related to some item overflowing and then just stretching the whole window over the device size. 

See:
https://github.com/user-attachments/assets/50779205-f017-47fa-966c-825256d564a3

**Comment by ann0see** [comment]
If VoiceOver is on, the app crashes as soon as the connection dialog is opened

**Comment by ann0see** [comment]
Since the latest build is no longer crashing: https://testflight.apple.com/join/xZRwhf63 as public TestFlight link.

**Comment by ann0see** [comment]
In the forseable future we'll be stuck as Qt5 is x86 only and we must update to qt6. However building with qt6 on the latest version gets to a black screen. This is unusable. 
Also apple requires us to build with the latest SDK for App Store Submission. So unless we find a fix iOS builds will no longer be possible

**Comment by ann0see** [comment]
(context)

 Root-cause analysis for iOS Qt6 white/black screen (#3406)

- Mixer board QGroupBox with `sizeConstraint=SetMinimumSize` in `clientdlgbase.ui` (verticalLayout_2) causes the layout minimum to exceed screen size, stretching the window overflow.
- `lblConnectToServer` long label without `wordWrap` produces a huge sizeHint, same overflow effect. `.hide()` does not fix (Qt6 quirk); only removing from `.ui` code works (maybe).
- Mobile defaults: `GD_DEFAULT=GD_SLIMFADER` on iOS/Android. `CClientDlg` is **not** maximized on iOS; iOS forces GUI+client.
- Same pattern suspect in `connectdlgbase.ui` (long labels, no wordWrap).

*Generated with assistance from AI. Content may not be reproducible or correct*

**Comment by ann0see** [comment]
@coderabbitai There's a bunch of issues with Qt6 - I think that it's quite hard to get this right. For now I could only see black screens when compiling with Qt6. What do you think? Can you debug this? Is it something Jamulus can solve, work around or do we need to have Qt implement a bug fix? Dig through the code.

**Comment by ann0see** [comment]
@coderabbitai

> The main client window is not maximized on iOS.

I thought @mcfnord  already tried something with the chat dialog and failed. https://github.com/jamulussoftware/jamulus/pull/3830 Also I think it's very sketchy to just see a fully black screen. Search for other iOS related issues in the repo to find out where the problem could lie... Yes, we can probably try to apply *some* of the suggestions. But is smells like a Qt6 bug.

**Comment by ann0see** [comment]
@coderabbitai - then try again. The issue we are facing that jamulus compiles but comes back with a black screen while compiling with latest Xcode and latest Qt6. This is a **black screen**. It might be related to OpenGL/Metal rendering or similar. It seems as it it is a hard bug to solve.
